[Tan Yi Guan] iP

tenebrius1 opened the pull request:
## Duke
> “Sometimes it pays to stay in bed on Monday, rather than spending the rest of the week debugging Monday’s code.” – Dan Salomon

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. download the `.jar` file from [here](https://github.com/tenebrius1/ip/releases/tag/v0.1).
2. run it using `java -jar ip.jar`.
3. add your tasks.
4. let it manage your tasks for you! 😉 

Best part? It is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke("./data/task.txt").run();
}
```

Commit:
Add Gradle support

tenebrius1 committed:
complete Level-1

tenebrius1 committed:
add more information to welcome message

tenebrius1 committed:
add javadocs

tenebrius1 committed:
complete Level-2

tenebrius1 committed:
complete Level-3

tenebrius1 committed:
add javadocs for Task.java and Command.java

tenebrius1 committed:
fix minor mistakes in JavaDocs

tenebrius1 committed:
complete Level-4

tenebrius1 committed:
update integration test

tenebrius1 committed:
complete Level-5

tenebrius1 committed:
add try-catch blocks

tenebrius1 committed:
complete Level-6

tenebrius1 committed:
update welcome message

tenebrius1 committed:
improve error detection

tenebrius1 committed:
Add functionality: Persisting data

tenebrius1 committed:
Change Task.java and subclasses: Remove getters

tenebrius1 committed:
Add functionality: Detect dateTime

tenebrius1 committed:
Merge branch 'branch-Level-7'

tenebrius1 committed:
Merge branch 'branch-Level-8'

tenebrius1 committed:
Restructure code: Split code into logical classes

tenebrius1 committed:
Add dateTime support for Event task

tenebrius1 committed:
Add subclasses of DukeException

tenebrius1 committed:
Code restructure: Group relavant classes into packages

tenebrius1 committed:
Add JUnit testing

tenebrius1 committed:
Add functionality: Filter by date

tenebrius1 committed:
Refactor code: Extract *Command classes

tenebrius1 committed:
Improve code quality: Add JavaDocs

tenebrius1 committed:
Improve code quality: Adhere to coding standards

tenebrius1 committed:
Add functionality: Search by keyword

tenebrius1 committed:
Merge branch 'branch-A-JavaDoc'

tenebrius1 committed:
Merge branch 'branch-A-CodingStandard'

tenebrius1 committed:
Merge branch 'branch-Level-9'

tenebrius1 committed:
Add JavaDoc for FindCommand.java

tenebrius1 committed:
Merge branch 'add-gradle-support'

tenebrius1 committed:
Integrate Gradle

tenebrius1 committed:
Merge branch 'branch-A-Gradle'

tenebrius1 committed:
Integrate Checkstyle

tenebrius1 committed:
Merge branch 'branch-A-Checkstyle'

tenebrius1 committed:
Feature: Add GUI using JavaFX

tenebrius1 committed:
Merge branch 'branch-Level-10'

tenebrius1 committed:
Fix: Make Duke auto close after 'exit' command

tenebrius1 committed:
Improvement: Schedule multiple todo with same command

luminousleek reviewed (commented):
I'm only supposed to comment on coding standard violations, but you have checkstyle implemented so that makes my life much easier :)

Great job overall on the iP though - your JavaFX implementation is a lot neater than mine

deetoomoo commented on src/main/java/duke/command/FilterCommand.java:
```diff
@@ -0,0 +1,46 @@
+
+        if (extractedTask.size() == 0) {
+            return "There are no tasks on this day.";
+        } else {
+            String output = String.format("Here are your tasks for this day:%s", Ui.LINE_SEPARATOR);
+
+            int count = 1;
+            for (Task t : extractedTask) {
+                output = output.concat(String.format("[%d]. %s", count++, t + Ui.LINE_SEPARATOR + "\t\t"));
+            }
```
Having the Ui class here introduces an unnecessary dependency. Consider decoupling?

deetoomoo commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,82 @@
+    @Override
+    public String toString() {
+        return String.format("[E]%s (at: %s)", super.toString(), this.dateTimeToString());
+    }
```
this. reference is not needed, honestly just commenting this because I have nothing else to add.. :-)

deetoomoo commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,146 @@
+            return "Todo command has to be followed by a task description!";
+        }
+
+        if (input.split(", ").length >= 2) {
```
Is this if statement needed? 🤔

deetoomoo reviewed (commented):
Good job overall on code quality! Just some minor comments :-)

tenebrius1 committed:
Add Java Assertions

tenebrius1 committed:
Improve code quality

There are several methods that do not follow the code quality standards specified in the textbook.

Refactoring the code improves code quality.

tenebrius1 committed:
Refactor code: Factor out magic strings

There are magic strings in the code base which does not comply with code quality standards.

Refactoring out magic strings would help improve code readability.

tenebrius1 committed:
Add streams support

tenebrius1 committed:
Merge pull request #2 from tenebrius1/branch-A-Assertions

Add Java Assertions

tenebrius1 committed:
Merge branch 'master' into branch-A-CodeQuality

tenebrius1 committed:
Merge pull request #4 from tenebrius1/branch-A-CodeQuality

Improve code quality

tenebrius1 committed:
Merge branch 'master' into branch-A-Streams

tenebrius1 committed:
Merge pull request #5 from tenebrius1/branch-A-Streams

Add streams support

tenebrius1 committed:
Fix bugs

tenebrius1 committed:
Add GitHub Actions support

tenebrius1 committed:
Fix errors caught by checkstyle test

tenebrius1 committed:
Improvement: Friendlier syntax support

The commands supported currently are too rigid.

A more flexible command syntax allows user to achieve the same task but with more efficiency using shorter aliases for commands.

tenebrius1 committed:
Merge pull request #6 from tenebrius1/branch-C-FriendlierSyntax

Improvement: Friendlier syntax support

tenebrius1 committed:
Improvement: Duplicate detection

tenebrius1 committed:
Merge branch 'branch-C-DetectDuplicates'

tenebrius1 committed:
Improvement: Better GUI

tenebrius1 committed:
Add User Guide

tenebrius1 committed:
Fix: Align picture to center

tenebrius1 committed:
Set theme jekyll-theme-modernist

tenebrius1 committed:
Feat: Ensure cross-platfornm compatibility

tenebrius1 committed:
Merge remote-tracking branch 'origin/master'

tenebrius1 committed:
Update _config.yml
